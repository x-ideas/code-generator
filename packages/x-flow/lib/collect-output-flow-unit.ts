/*
 * @name:
 * description:
 *  用于收集所有子flow unit的output的数据，统一传递给下一个flow unit
 */

import { XGroupedFlowUnit } from './grouped-flow-unit';

export class CollectOutputFlowUnit extends XGroupedFlowUnit {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
  async doWork(data: any): Promise<any[]> {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const result: any[] = [];

    let nextData = data;
    for (const unit of this._units) {
      nextData = await unit.doWork(nextData);
      result.push(nextData);
    }

    return result;
  }
}
