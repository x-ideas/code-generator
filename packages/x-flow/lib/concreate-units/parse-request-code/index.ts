export * from './parser-request-code-unit';
