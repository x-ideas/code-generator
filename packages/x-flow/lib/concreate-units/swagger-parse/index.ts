export * from './swagger-parser-flow-unit';
