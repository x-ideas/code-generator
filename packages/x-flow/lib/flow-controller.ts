import { XFlowUnit } from './flow-unit';

type FuncType = (...args: unknown[]) => unknown;

function compose(funcs: FuncType[]): FuncType {
  return funcs.reduce((accum, current) => {
    return async function (data: unknown) {
      const nextData: any = await accum(data);
      return current(nextData);
    };
  });
}

export class XFlowController {
  private _units: XFlowUnit[];

  constructor() {
    this._units = [];
  }

  addUnit(unit: XFlowUnit): void {
    this._units.push(unit);
  }

  async run(data?: any): Promise<any> {
    const final = compose(this._units.map(unit => unit.doWork.bind(unit)));

    const result = final(data);
    return result;
  }
}
