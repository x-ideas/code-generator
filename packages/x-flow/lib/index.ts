export * from './concreate-units/generate-service-request';
