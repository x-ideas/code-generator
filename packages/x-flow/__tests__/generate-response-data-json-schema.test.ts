import { CollectOutputFlowUnit } from '../lib/collect-output-flow-unit';
import { ParseRequestCodeFlowUnit } from '../lib/concreate-units/parse-request-code';
import { SpecialOutputFlowUnit } from '../lib/concreate-units/specify-output';
import { SwaggerParseFlowUnit } from '../lib/concreate-units/swagger-parse';
import { XFlowController } from '../lib/flow-controller';

import OpenAPIData from './assets/openAPI.json';
import { GenerateResponseDataSchemaFlowUnit } from '../lib/concreate-units/generate-response-data-schema';
import { JSONSchema4 } from 'json-schema';
import { GenerateRequestParamsJsonSchemaFlowUnit } from '../lib/concreate-units/generate-request-params-json-schema';

describe('测试Generate Response Data JsonSchemaFlowUnit', () => {
  test('非list', async () => {
    const fc = new XFlowController();

    const collectOutputUnit = new CollectOutputFlowUnit();

    const outputCodeUnit = new SpecialOutputFlowUnit('3610115');
    const outputOpenAPIUnit = new SpecialOutputFlowUnit(OpenAPIData);
    collectOutputUnit.addUnit(outputCodeUnit);
    collectOutputUnit.addUnit(outputOpenAPIUnit);

    const codeParseUnit = new ParseRequestCodeFlowUnit();
    const swaggerParseUnit = new SwaggerParseFlowUnit();

    const frpUnit = new GenerateResponseDataSchemaFlowUnit();

    fc.addUnit(collectOutputUnit);
    fc.addUnit(codeParseUnit);
    fc.addUnit(swaggerParseUnit);
    fc.addUnit(frpUnit);

    const result: JSONSchema4 = await fc.run();
    expect(result).toMatchSnapshot();
  });

  test('list', async () => {
    const fc = new XFlowController();

    const collectOutputUnit = new CollectOutputFlowUnit();

    const outputCodeUnit = new SpecialOutputFlowUnit('3610401');
    const outputOpenAPIUnit = new SpecialOutputFlowUnit(OpenAPIData);
    collectOutputUnit.addUnit(outputCodeUnit);
    collectOutputUnit.addUnit(outputOpenAPIUnit);

    const codeParseUnit = new ParseRequestCodeFlowUnit();
    const swaggerParseUnit = new SwaggerParseFlowUnit();

    const frpUnit = new GenerateResponseDataSchemaFlowUnit();

    fc.addUnit(collectOutputUnit);
    fc.addUnit(codeParseUnit);
    fc.addUnit(swaggerParseUnit);
    fc.addUnit(frpUnit);

    const result: JSONSchema4 = await fc.run();
    expect(result).toMatchSnapshot();
  });

  test('循环-3610116', async () => {
    const fc = new XFlowController();

    const collectOutputUnit = new CollectOutputFlowUnit();

    const outputCodeUnit = new SpecialOutputFlowUnit('3610116');
    const outputOpenAPIUnit = new SpecialOutputFlowUnit(OpenAPIData);
    collectOutputUnit.addUnit(outputCodeUnit);
    collectOutputUnit.addUnit(outputOpenAPIUnit);

    const codeParseUnit = new ParseRequestCodeFlowUnit();
    const swaggerParseUnit = new SwaggerParseFlowUnit();

    const frpUnit = new GenerateResponseDataSchemaFlowUnit();

    fc.addUnit(collectOutputUnit);
    fc.addUnit(codeParseUnit);
    fc.addUnit(swaggerParseUnit);
    fc.addUnit(frpUnit);

    const result: JSONSchema4 = await fc.run();
    expect(result).toMatchSnapshot();
  });

  test('只包含body+header', async () => {
    const fc = new XFlowController();

    const collectOutputUnit = new CollectOutputFlowUnit();

    const outputCodeUnit = new SpecialOutputFlowUnit('3610116');
    const outputOpenAPIUnit = new SpecialOutputFlowUnit(OpenAPIData);
    collectOutputUnit.addUnit(outputCodeUnit);
    collectOutputUnit.addUnit(outputOpenAPIUnit);

    const codeParseUnit = new ParseRequestCodeFlowUnit();
    const swaggerParseUnit = new SwaggerParseFlowUnit();

    const frpUnit = new GenerateResponseDataSchemaFlowUnit();

    fc.addUnit(collectOutputUnit);
    fc.addUnit(codeParseUnit);
    fc.addUnit(swaggerParseUnit);
    fc.addUnit(frpUnit);

    const result = await fc.run();
    expect(result).toMatchSnapshot();
  });

  test('不包含body和path', async () => {
    const fc = new XFlowController();

    const collectOutputUnit = new CollectOutputFlowUnit();

    const outputCodeUnit = new SpecialOutputFlowUnit('3610401');
    const outputOpenAPIUnit = new SpecialOutputFlowUnit(OpenAPIData);
    collectOutputUnit.addUnit(outputCodeUnit);
    collectOutputUnit.addUnit(outputOpenAPIUnit);

    const codeParseUnit = new ParseRequestCodeFlowUnit();
    const swaggerParseUnit = new SwaggerParseFlowUnit();

    const frpUnit = new GenerateResponseDataSchemaFlowUnit();

    fc.addUnit(collectOutputUnit);
    fc.addUnit(codeParseUnit);
    fc.addUnit(swaggerParseUnit);
    fc.addUnit(frpUnit);

    const result = await fc.run();
    expect(result).toMatchSnapshot();
  });

  test('分页list响应', async () => {
    const fc = new XFlowController();

    const collectOutputUnit = new CollectOutputFlowUnit();

    const outputCodeUnit = new SpecialOutputFlowUnit('3783510');
    const outputOpenAPIUnit = new SpecialOutputFlowUnit(OpenAPIData);
    collectOutputUnit.addUnit(outputCodeUnit);
    collectOutputUnit.addUnit(outputOpenAPIUnit);

    const codeParseUnit = new ParseRequestCodeFlowUnit();
    const swaggerParseUnit = new SwaggerParseFlowUnit();

    const frpUnit = new GenerateResponseDataSchemaFlowUnit();

    fc.addUnit(collectOutputUnit);
    fc.addUnit(codeParseUnit);
    fc.addUnit(swaggerParseUnit);
    fc.addUnit(frpUnit);

    const result = await fc.run();
    expect(result).toMatchSnapshot();
  });
});
