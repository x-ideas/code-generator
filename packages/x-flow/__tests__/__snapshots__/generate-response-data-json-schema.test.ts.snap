// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试Generate Response Data JsonSchemaFlowUnit list 1`] = `
Object {
  "dataSchema": Object {
    "description": undefined,
    "items": Object {
      "items": Object {
        "description": "补货需求单，列表用",
        "properties": Object {
          "apply_no": Object {
            "default": undefined,
            "description": "需求单号",
            "required": undefined,
            "type": "string",
          },
          "apply_type": Object {
            "default": undefined,
            "description": "补货类型,10=旧版采购补货,20=总仓自动补货,30=DC仓采购补货(手动)",
            "required": undefined,
            "type": "integer",
          },
          "date_part_plan": Object {
            "default": undefined,
            "description": "计划到货时间段，10=白天，20=晚上",
            "required": undefined,
            "type": "integer",
          },
          "date_plan": Object {
            "default": undefined,
            "description": "计划到货日期",
            "required": undefined,
            "type": "integer",
          },
          "dc_abbr_name": Object {
            "default": undefined,
            "description": "总仓简称",
            "required": undefined,
            "type": "string",
          },
          "dc_id": Object {
            "default": undefined,
            "description": "总仓id",
            "required": undefined,
            "type": "string",
          },
          "dc_name": Object {
            "default": undefined,
            "description": "总仓名称",
            "required": undefined,
            "type": "string",
          },
          "id": Object {
            "default": undefined,
            "description": "id",
            "required": undefined,
            "type": "string",
          },
          "is_delete": Object {
            "default": undefined,
            "description": "是否删除",
            "required": undefined,
            "type": "boolean",
          },
          "item_count_plan": Object {
            "default": undefined,
            "description": "计划品项数量",
            "required": undefined,
            "type": "integer",
          },
          "item_count_real": Object {
            "default": undefined,
            "description": "实际品项数量",
            "required": undefined,
            "type": "integer",
          },
          "order_id": Object {
            "default": undefined,
            "description": "补货订单id",
            "required": undefined,
            "type": "string",
          },
          "status": Object {
            "default": undefined,
            "description": "状态，10=待发送，20=待发货，30=待收货，40=到货中，50=已完成",
            "required": undefined,
            "type": "integer",
          },
          "store_abbr_name": Object {
            "default": undefined,
            "description": "门店简称",
            "required": undefined,
            "type": "string",
          },
          "store_code": Object {
            "default": undefined,
            "description": "门店code",
            "required": undefined,
            "type": "integer",
          },
          "store_id": Object {
            "default": undefined,
            "description": "门店id",
            "required": undefined,
            "type": "string",
          },
          "store_name": Object {
            "default": undefined,
            "description": "门店名称",
            "required": undefined,
            "type": "string",
          },
          "time_create": Object {
            "default": undefined,
            "description": "创建时间",
            "required": undefined,
            "type": "integer",
          },
          "time_plan": Object {
            "default": undefined,
            "description": "计划到货时间",
            "required": undefined,
            "type": "integer",
          },
          "time_real": Object {
            "default": undefined,
            "description": "实际到货时间",
            "required": undefined,
            "type": "integer",
          },
          "time_update": Object {
            "default": undefined,
            "description": "更新时间",
            "required": undefined,
            "type": "integer",
          },
          "user_id_update": Object {
            "default": undefined,
            "description": "更新人id",
            "required": undefined,
            "type": "string",
          },
          "user_name_update": Object {
            "default": undefined,
            "description": "更新人名称",
            "required": undefined,
            "type": "string",
          },
        },
        "type": "object",
      },
      "type": "array",
    },
    "type": "array",
  },
  "isPageList": true,
}
`;

exports[`测试Generate Response Data JsonSchemaFlowUnit 不包含body和path 1`] = `
Object {
  "dataSchema": Object {
    "description": undefined,
    "items": Object {
      "items": Object {
        "description": "补货需求单，列表用",
        "properties": Object {
          "apply_no": Object {
            "default": undefined,
            "description": "需求单号",
            "required": undefined,
            "type": "string",
          },
          "apply_type": Object {
            "default": undefined,
            "description": "补货类型,10=旧版采购补货,20=总仓自动补货,30=DC仓采购补货(手动)",
            "required": undefined,
            "type": "integer",
          },
          "date_part_plan": Object {
            "default": undefined,
            "description": "计划到货时间段，10=白天，20=晚上",
            "required": undefined,
            "type": "integer",
          },
          "date_plan": Object {
            "default": undefined,
            "description": "计划到货日期",
            "required": undefined,
            "type": "integer",
          },
          "dc_abbr_name": Object {
            "default": undefined,
            "description": "总仓简称",
            "required": undefined,
            "type": "string",
          },
          "dc_id": Object {
            "default": undefined,
            "description": "总仓id",
            "required": undefined,
            "type": "string",
          },
          "dc_name": Object {
            "default": undefined,
            "description": "总仓名称",
            "required": undefined,
            "type": "string",
          },
          "id": Object {
            "default": undefined,
            "description": "id",
            "required": undefined,
            "type": "string",
          },
          "is_delete": Object {
            "default": undefined,
            "description": "是否删除",
            "required": undefined,
            "type": "boolean",
          },
          "item_count_plan": Object {
            "default": undefined,
            "description": "计划品项数量",
            "required": undefined,
            "type": "integer",
          },
          "item_count_real": Object {
            "default": undefined,
            "description": "实际品项数量",
            "required": undefined,
            "type": "integer",
          },
          "order_id": Object {
            "default": undefined,
            "description": "补货订单id",
            "required": undefined,
            "type": "string",
          },
          "status": Object {
            "default": undefined,
            "description": "状态，10=待发送，20=待发货，30=待收货，40=到货中，50=已完成",
            "required": undefined,
            "type": "integer",
          },
          "store_abbr_name": Object {
            "default": undefined,
            "description": "门店简称",
            "required": undefined,
            "type": "string",
          },
          "store_code": Object {
            "default": undefined,
            "description": "门店code",
            "required": undefined,
            "type": "integer",
          },
          "store_id": Object {
            "default": undefined,
            "description": "门店id",
            "required": undefined,
            "type": "string",
          },
          "store_name": Object {
            "default": undefined,
            "description": "门店名称",
            "required": undefined,
            "type": "string",
          },
          "time_create": Object {
            "default": undefined,
            "description": "创建时间",
            "required": undefined,
            "type": "integer",
          },
          "time_plan": Object {
            "default": undefined,
            "description": "计划到货时间",
            "required": undefined,
            "type": "integer",
          },
          "time_real": Object {
            "default": undefined,
            "description": "实际到货时间",
            "required": undefined,
            "type": "integer",
          },
          "time_update": Object {
            "default": undefined,
            "description": "更新时间",
            "required": undefined,
            "type": "integer",
          },
          "user_id_update": Object {
            "default": undefined,
            "description": "更新人id",
            "required": undefined,
            "type": "string",
          },
          "user_name_update": Object {
            "default": undefined,
            "description": "更新人名称",
            "required": undefined,
            "type": "string",
          },
        },
        "type": "object",
      },
      "type": "array",
    },
    "type": "array",
  },
  "isPageList": true,
}
`;

exports[`测试Generate Response Data JsonSchemaFlowUnit 分页list响应 1`] = `
Object {
  "dataSchema": Object {
    "description": undefined,
    "items": Object {
      "description": "AfterSaleBoxListVO",
      "properties": Object {
        "box_mark": Object {
          "default": undefined,
          "description": "箱号，如1/3",
          "required": undefined,
          "type": "string",
        },
        "box_num": Object {
          "default": undefined,
          "description": "退货箱编号",
          "required": undefined,
          "type": "string",
        },
        "box_status": Object {
          "default": undefined,
          "description": "处理状态，10=待装车，20=已装车，30=待接收，40=已接收，50=已存放，60=待取货，70=已取货，80=已退换",
          "required": undefined,
          "type": "integer",
        },
        "id": Object {
          "default": undefined,
          "description": "退货箱id",
          "required": undefined,
          "type": "string",
        },
        "service_id": Object {
          "default": undefined,
          "description": "售后单id",
          "required": undefined,
          "type": "string",
        },
        "service_num": Object {
          "default": undefined,
          "description": "售后单编号",
          "required": undefined,
          "type": "string",
        },
        "service_type": Object {
          "default": undefined,
          "description": "退换类型，0=换货,20=退货",
          "required": undefined,
          "type": "integer",
        },
        "shelf_code": Object {
          "default": undefined,
          "description": "存放库位",
          "required": undefined,
          "type": "string",
        },
        "status_after_sale": Object {
          "default": undefined,
          "description": "售后方式，10=门店处理，20=总仓处理",
          "required": undefined,
          "type": "integer",
        },
        "supplier_code": Object {
          "default": undefined,
          "description": "供应商编号",
          "required": undefined,
          "type": "integer",
        },
        "supplier_id": Object {
          "default": undefined,
          "description": "供应商ID",
          "required": undefined,
          "type": "string",
        },
        "supplier_name": Object {
          "default": undefined,
          "description": "供应商名称",
          "required": undefined,
          "type": "string",
        },
        "supplier_name_abbr": Object {
          "default": undefined,
          "description": "供应商简称",
          "required": undefined,
          "type": "string",
        },
      },
      "type": "object",
    },
    "type": "array",
  },
  "isPageList": true,
}
`;

exports[`测试Generate Response Data JsonSchemaFlowUnit 只包含body+header 1`] = `
Object {
  "dataSchema": Object {
    "description": undefined,
    "properties": Object {
      "apply_count": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "apply_count_finish": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "apply_type": Object {
        "default": undefined,
        "description": "补货类型：20=总仓自动补货，30=DC仓采购补货",
        "required": undefined,
        "type": "integer",
      },
      "batch_count": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "batch_type": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "date_part_plan": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "date_plan": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "dc_id": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "dc_name": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "has_deleted": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "boolean",
      },
      "id": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "item_count": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "num": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "overview": Object {
        "description": "概览信息",
        "properties": Object {
          "base_info": Object {
            "description": "基本信息",
            "properties": Object {
              "collect_count": Object {
                "default": undefined,
                "description": "集货数",
                "required": undefined,
                "type": "integer",
              },
              "item_count": Object {
                "default": undefined,
                "description": "品项数",
                "required": undefined,
                "type": "integer",
              },
              "order_count": Object {
                "default": undefined,
                "description": "波次数",
                "required": undefined,
                "type": "integer",
              },
              "order_list": Object {
                "description": "波次列表",
                "items": undefined,
                "type": "array",
              },
              "store_count": Object {
                "default": undefined,
                "description": "门店数",
                "required": undefined,
                "type": "integer",
              },
              "time_group": Object {
                "default": undefined,
                "description": "小时",
                "required": undefined,
                "type": "integer",
              },
              "volume": Object {
                "default": undefined,
                "description": "体积(m³)",
                "required": undefined,
                "type": "integer",
              },
              "weight": Object {
                "default": undefined,
                "description": "重量(g)",
                "required": undefined,
                "type": "integer",
              },
            },
            "type": "object",
          },
          "process": Object {
            "description": "进度动态",
            "properties": Object {
              "collect_progress": Object {
                "default": undefined,
                "description": "集货进度",
                "required": undefined,
                "type": "number",
              },
              "collect_progress_relative": Object {
                "default": undefined,
                "description": "相对集货进度",
                "required": undefined,
                "type": "number",
              },
              "collect_sum_correct": Object {
                "default": undefined,
                "description": "校正集货数总和",
                "required": undefined,
                "type": "integer",
              },
              "collect_sum_pending": Object {
                "default": undefined,
                "description": "待处理集货数总和",
                "required": undefined,
                "type": "integer",
              },
              "collect_sum_plan": Object {
                "default": undefined,
                "description": "计划集货数总和",
                "required": undefined,
                "type": "integer",
              },
              "collect_sum_real": Object {
                "default": undefined,
                "description": "实际集货数总和",
                "required": undefined,
                "type": "integer",
              },
              "delivery_progress": Object {
                "default": undefined,
                "description": "出货进度",
                "required": undefined,
                "type": "number",
              },
              "delivery_progress_relative": Object {
                "default": undefined,
                "description": "相对出货进度",
                "required": undefined,
                "type": "number",
              },
              "delivery_sum_real": Object {
                "default": undefined,
                "description": "实际出货数总和",
                "required": undefined,
                "type": "integer",
              },
              "distribute_progress": Object {
                "default": undefined,
                "description": "分货进度",
                "required": undefined,
                "type": "number",
              },
              "distribute_progress_relative": Object {
                "default": undefined,
                "description": "相对分货进度",
                "required": undefined,
                "type": "number",
              },
              "distribute_sum_real": Object {
                "default": undefined,
                "description": "实际分货数总和",
                "required": undefined,
                "type": "integer",
              },
              "order_progress": Object {
                "default": undefined,
                "description": "波次创建进度",
                "required": undefined,
                "type": "number",
              },
              "receive_progress": Object {
                "default": undefined,
                "description": "收货进度",
                "required": undefined,
                "type": "number",
              },
              "receive_progress_relative": Object {
                "default": undefined,
                "description": "相对收货进度",
                "required": undefined,
                "type": "number",
              },
              "receive_sum_real": Object {
                "default": undefined,
                "description": "实际入库数总和",
                "required": undefined,
                "type": "integer",
              },
            },
            "type": "object",
          },
        },
        "type": "object",
      },
      "progress": Object {
        "description": "进度",
        "properties": Object {
          "collect_progress": Object {
            "default": undefined,
            "description": "集货进度",
            "required": undefined,
            "type": "number",
          },
          "collect_progress_relative": Object {
            "default": undefined,
            "description": "相对集货进度",
            "required": undefined,
            "type": "number",
          },
          "collect_sum_correct": Object {
            "default": undefined,
            "description": "校正集货数总和",
            "required": undefined,
            "type": "integer",
          },
          "collect_sum_pending": Object {
            "default": undefined,
            "description": "待处理集货数总和",
            "required": undefined,
            "type": "integer",
          },
          "collect_sum_plan": Object {
            "default": undefined,
            "description": "计划集货数总和",
            "required": undefined,
            "type": "integer",
          },
          "collect_sum_real": Object {
            "default": undefined,
            "description": "实际集货数总和",
            "required": undefined,
            "type": "integer",
          },
          "delivery_progress": Object {
            "default": undefined,
            "description": "出货进度",
            "required": undefined,
            "type": "number",
          },
          "delivery_progress_relative": Object {
            "default": undefined,
            "description": "相对出货进度",
            "required": undefined,
            "type": "number",
          },
          "delivery_sum_real": Object {
            "default": undefined,
            "description": "实际出货数总和",
            "required": undefined,
            "type": "integer",
          },
          "distribute_progress": Object {
            "default": undefined,
            "description": "分货进度",
            "required": undefined,
            "type": "number",
          },
          "distribute_progress_relative": Object {
            "default": undefined,
            "description": "相对分货进度",
            "required": undefined,
            "type": "number",
          },
          "distribute_sum_real": Object {
            "default": undefined,
            "description": "实际分货数总和",
            "required": undefined,
            "type": "integer",
          },
          "order_progress": Object {
            "default": undefined,
            "description": "波次创建进度",
            "required": undefined,
            "type": "number",
          },
          "receive_progress": Object {
            "default": undefined,
            "description": "收货进度",
            "required": undefined,
            "type": "number",
          },
          "receive_progress_relative": Object {
            "default": undefined,
            "description": "相对收货进度",
            "required": undefined,
            "type": "number",
          },
          "receive_sum_real": Object {
            "default": undefined,
            "description": "实际入库数总和",
            "required": undefined,
            "type": "integer",
          },
        },
        "type": "object",
      },
      "remark": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "status": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "time_create": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "time_plan": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "time_update": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "user_id_create": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "user_id_update": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "user_name_create": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "user_name_update": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
    },
    "type": "object",
  },
  "isPageList": false,
}
`;

exports[`测试Generate Response Data JsonSchemaFlowUnit 循环-3610116 1`] = `
Object {
  "dataSchema": Object {
    "description": undefined,
    "properties": Object {
      "apply_count": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "apply_count_finish": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "apply_type": Object {
        "default": undefined,
        "description": "补货类型：20=总仓自动补货，30=DC仓采购补货",
        "required": undefined,
        "type": "integer",
      },
      "batch_count": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "batch_type": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "date_part_plan": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "date_plan": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "dc_id": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "dc_name": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "has_deleted": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "boolean",
      },
      "id": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "item_count": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "num": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "overview": Object {
        "description": "概览信息",
        "properties": Object {
          "base_info": Object {
            "description": "基本信息",
            "properties": Object {
              "collect_count": Object {
                "default": undefined,
                "description": "集货数",
                "required": undefined,
                "type": "integer",
              },
              "item_count": Object {
                "default": undefined,
                "description": "品项数",
                "required": undefined,
                "type": "integer",
              },
              "order_count": Object {
                "default": undefined,
                "description": "波次数",
                "required": undefined,
                "type": "integer",
              },
              "order_list": Object {
                "description": "波次列表",
                "items": undefined,
                "type": "array",
              },
              "store_count": Object {
                "default": undefined,
                "description": "门店数",
                "required": undefined,
                "type": "integer",
              },
              "time_group": Object {
                "default": undefined,
                "description": "小时",
                "required": undefined,
                "type": "integer",
              },
              "volume": Object {
                "default": undefined,
                "description": "体积(m³)",
                "required": undefined,
                "type": "integer",
              },
              "weight": Object {
                "default": undefined,
                "description": "重量(g)",
                "required": undefined,
                "type": "integer",
              },
            },
            "type": "object",
          },
          "process": Object {
            "description": "进度动态",
            "properties": Object {
              "collect_progress": Object {
                "default": undefined,
                "description": "集货进度",
                "required": undefined,
                "type": "number",
              },
              "collect_progress_relative": Object {
                "default": undefined,
                "description": "相对集货进度",
                "required": undefined,
                "type": "number",
              },
              "collect_sum_correct": Object {
                "default": undefined,
                "description": "校正集货数总和",
                "required": undefined,
                "type": "integer",
              },
              "collect_sum_pending": Object {
                "default": undefined,
                "description": "待处理集货数总和",
                "required": undefined,
                "type": "integer",
              },
              "collect_sum_plan": Object {
                "default": undefined,
                "description": "计划集货数总和",
                "required": undefined,
                "type": "integer",
              },
              "collect_sum_real": Object {
                "default": undefined,
                "description": "实际集货数总和",
                "required": undefined,
                "type": "integer",
              },
              "delivery_progress": Object {
                "default": undefined,
                "description": "出货进度",
                "required": undefined,
                "type": "number",
              },
              "delivery_progress_relative": Object {
                "default": undefined,
                "description": "相对出货进度",
                "required": undefined,
                "type": "number",
              },
              "delivery_sum_real": Object {
                "default": undefined,
                "description": "实际出货数总和",
                "required": undefined,
                "type": "integer",
              },
              "distribute_progress": Object {
                "default": undefined,
                "description": "分货进度",
                "required": undefined,
                "type": "number",
              },
              "distribute_progress_relative": Object {
                "default": undefined,
                "description": "相对分货进度",
                "required": undefined,
                "type": "number",
              },
              "distribute_sum_real": Object {
                "default": undefined,
                "description": "实际分货数总和",
                "required": undefined,
                "type": "integer",
              },
              "order_progress": Object {
                "default": undefined,
                "description": "波次创建进度",
                "required": undefined,
                "type": "number",
              },
              "receive_progress": Object {
                "default": undefined,
                "description": "收货进度",
                "required": undefined,
                "type": "number",
              },
              "receive_progress_relative": Object {
                "default": undefined,
                "description": "相对收货进度",
                "required": undefined,
                "type": "number",
              },
              "receive_sum_real": Object {
                "default": undefined,
                "description": "实际入库数总和",
                "required": undefined,
                "type": "integer",
              },
            },
            "type": "object",
          },
        },
        "type": "object",
      },
      "progress": Object {
        "description": "进度",
        "properties": Object {
          "collect_progress": Object {
            "default": undefined,
            "description": "集货进度",
            "required": undefined,
            "type": "number",
          },
          "collect_progress_relative": Object {
            "default": undefined,
            "description": "相对集货进度",
            "required": undefined,
            "type": "number",
          },
          "collect_sum_correct": Object {
            "default": undefined,
            "description": "校正集货数总和",
            "required": undefined,
            "type": "integer",
          },
          "collect_sum_pending": Object {
            "default": undefined,
            "description": "待处理集货数总和",
            "required": undefined,
            "type": "integer",
          },
          "collect_sum_plan": Object {
            "default": undefined,
            "description": "计划集货数总和",
            "required": undefined,
            "type": "integer",
          },
          "collect_sum_real": Object {
            "default": undefined,
            "description": "实际集货数总和",
            "required": undefined,
            "type": "integer",
          },
          "delivery_progress": Object {
            "default": undefined,
            "description": "出货进度",
            "required": undefined,
            "type": "number",
          },
          "delivery_progress_relative": Object {
            "default": undefined,
            "description": "相对出货进度",
            "required": undefined,
            "type": "number",
          },
          "delivery_sum_real": Object {
            "default": undefined,
            "description": "实际出货数总和",
            "required": undefined,
            "type": "integer",
          },
          "distribute_progress": Object {
            "default": undefined,
            "description": "分货进度",
            "required": undefined,
            "type": "number",
          },
          "distribute_progress_relative": Object {
            "default": undefined,
            "description": "相对分货进度",
            "required": undefined,
            "type": "number",
          },
          "distribute_sum_real": Object {
            "default": undefined,
            "description": "实际分货数总和",
            "required": undefined,
            "type": "integer",
          },
          "order_progress": Object {
            "default": undefined,
            "description": "波次创建进度",
            "required": undefined,
            "type": "number",
          },
          "receive_progress": Object {
            "default": undefined,
            "description": "收货进度",
            "required": undefined,
            "type": "number",
          },
          "receive_progress_relative": Object {
            "default": undefined,
            "description": "相对收货进度",
            "required": undefined,
            "type": "number",
          },
          "receive_sum_real": Object {
            "default": undefined,
            "description": "实际入库数总和",
            "required": undefined,
            "type": "integer",
          },
        },
        "type": "object",
      },
      "remark": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "status": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "time_create": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "time_plan": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "time_update": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "integer",
      },
      "user_id_create": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "user_id_update": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "user_name_create": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
      "user_name_update": Object {
        "default": undefined,
        "description": "",
        "required": undefined,
        "type": "string",
      },
    },
    "type": "object",
  },
  "isPageList": false,
}
`;

exports[`测试Generate Response Data JsonSchemaFlowUnit 非list 1`] = `
Object {
  "dataSchema": Object {
    "description": undefined,
    "properties": Object {},
    "type": "object",
  },
  "isPageList": false,
}
`;
